File: beecrowd_exercicios_em_Ruby/mathematics/base_32.rb
def to_32(x)
  rslt = ''

  return 0 if x == 0

  while x != 0
    if x % 32 > 9
      rslt += (x%32 + 55).chr
    else
      rslt += (x % 32).to_s
    end
    x /= 32
  end

  rslt.reverse!

end



loop do

  n = gets.to_i
  puts to_32(n)
  break if n == 0

end

File: beecrowd_exercicios_em_Python/mathematics/base_32.py
def dec_to_32(x):

    rslt = ""

    while x != 0:
        if x % 32 > 9:
            rslt += chr(x % 32 + 55)
        else:
            rslt += str(x % 32)

        x //= 32

    return rslt[::-1]


n = int(input())

while n != 0:
    print(dec_to_32(n))
    n = int(input())

print(0)
